client/src/components/Sidebar/UserList/index.tsx

```diff
@@ -1,15 +1,26 @@
-import React, { useContext } from 'react'
-import { UserInfoContext } from "@/contexts";
+import React, { useContext, useRef, useState } from 'react'
+import { UserInfoContext, SocketContext } from "@/contexts";
 import { UserInfo } from "@/contexts/UserInfoContext";
 import AvatarDOM from "@/components/AvatarCanvas/AvatarDOM"
+import { SIOChannel } from "@/contexts/SocketIOContext";
 
 const UserList = () => {
   // TODO: Need to have selfUserInfo here as well
   const { userInfos } = useContext(UserInfoContext);
+  const userInfosRef = useRef(userInfos);
+  const { socket } = useContext(SocketContext);
+  
+  const editName = () => {
+    const newName = prompt('Enter new name');
+    if (newName) {
+      socket.emit(SIOChannel.EDIT_USER_NAME, newName);
+    }
+  };
+
   // FIXME: Styling of user list
   return (
     <>
-      {(Object.values(userInfos) as UserInfo[]).map((userInfo: UserInfo, index: number) => (
+      {(Object.values(userInfosRef.current) as UserInfo[]).map((userInfo: UserInfo, index: number) => (
```

Comment:
Sorry my bad! I realized the bug was unrelated to using ref. I think what you had before here without the useRef was fine.

Reply:
Still need to revert this back to `userInfos` (and remove userInfosRef, and the useRef import) sorry about that!